src/dmclock_server.h

```diff
@@ -757,12 +759,14 @@ namespace crimson {
       Duration                  idle_age;
       Duration                  erase_age;
       Duration                  check_time;
-      std::deque<MarkPoint>     clean_mark_points;
+      std::deque<MarkPoint>     idle_erase_mark_points;
 
-      // NB: All threads declared at end, so they're destructed first!
+      using IdleEraseListener = std::function<void(const C&)>;
+      std::set<const IdleEraseListener*> idle_erase_listeners;
```

Comment:
it doesn't seem safe to store these functions by pointer - especially when they're passed to `add_idle_erase_listener()` as a const reference. that means it's perfectly valid to call with a lambda, which we'd store as a pointer to a temporary object:
```c++
  q.add_idle_erase_listener([] (const C& c) {});
```
it looks like it was done this way so you could implement `remove_idle_erase_listener()` - but is this kind of dynamic registration really necessary? consider treating this similar to `ClientInfoFunc`, and take a single optional callback as a constructor argument. if the caller really needs dynamic registration, they could implement it themselves inside their callback. does that make sense?